File: frontend/src/services/authService.js
import axiosClient from "@/api/axiosClient";
import { ENDPOINTS } from "@/api/endpoints";

class AuthService {

    async login(data) {

        const response = await axiosClient.post(
            ENDPOINTS.AUTH.LOGIN,
            data
        );

        return response.data;

    }

    async logout() {

        const response = await axiosClient.post(
            ENDPOINTS.AUTH.LOGOUT
        );

        return response.data;

    }

    async sendOtp(data) {

        const response = await axiosClient.post(
            ENDPOINTS.AUTH.SEND_OTP,
            data
        );

        return response.data;

    }

    async verifyOtp(data) {

        const response = await axiosClient.post(
            ENDPOINTS.AUTH.VERIFY_OTP,
            data
        );

        return response.data;

    }

    async me() {

        const response = await axiosClient.get(
            ENDPOINTS.AUTH.ME
        );

        return response.data;

    }

}

export default new AuthService();

File: Job Portal/job-portal-system/services/job-portal-user-service/src/main/java/com/project/referral/auth/service/AuthService.java
package com.project.referral.auth.service;

import com.project.referral.auth.dto.LoginRequest;
import com.project.referral.auth.dto.LoginResponse;
import com.project.referral.auth.dto.SendOtpRequest;
import com.project.referral.auth.dto.SignupResponse;
import com.project.referral.auth.dto.VerifyOtpRequest;

public interface AuthService {

    LoginResponse login(LoginRequest request);

    void sendOtp(SendOtpRequest request);

    SignupResponse verifyOtp(VerifyOtpRequest request);

    void resendOtp(String email);
}